src/Compilers/Core/Portable/Syntax/SyntaxTriviaList.cs

```diff
@@ -397,7 +397,7 @@ public SyntaxTriviaList ReplaceRange(SyntaxTrivia triviaInList, IEnumerable<Synt
                 var list = this.ToList();
                 list.RemoveAt(index);
                 list.InsertRange(index, newTrivia);
-                return new SyntaxTriviaList(default(SyntaxToken), GreenNode.CreateList(list.Select(n => n.RequiredUnderlyingNode)), 0, 0);
+                return new SyntaxTriviaList(default(SyntaxToken), GreenNode.CreateList(list, n => n.RequiredUnderlyingNode), 0, 0);
```

Comment:
looks like mosts callers pass in a List.  is there a need to specify IReadOnlyList rather than optmizing for that concrete type?